core/server/web/shared/middlewares/custom-redirects.js

```diff
@@ -15,42 +14,22 @@ const _private = {};
 
 let customRedirectsRouter;
 
+const getRedirectsFilePath = () => {
+    const yamlPath = path.join(config.getContentPath('data'), `redirects.yaml`);
+    const jsonPath = path.join(config.getContentPath('data'), `redirects.json`);
+
+    return fs.existsSync(yamlPath) ? yamlPath : jsonPath;
+};
+
 _private.registerRoutes = () => {
     debug('redirects loading');
 
     customRedirectsRouter = express.Router('redirects');
 
-    let redirects = [];
-
     try {
-        if (fs.existsSync(path.join(config.getContentPath('data'), 'redirects.yaml'))) {
-            let configYaml = yaml.safeLoad(fs.readFileSync(path.join(config.getContentPath('data'), 'redirects.yaml'), 'utf-8'));
-
-            /**
-             * 302: Temporary redirects
-             */
-            for (const redirect in configYaml['302']) {
-                redirects.push({
-                    from: redirect,
-                    to: configYaml['302'][redirect],
-                    permanent: false
-                });
-            }
-
-            /**
-             * 301: Permanent redirects
-             */
-            for (const redirect in configYaml['301']) {
-                redirects.push({
-                    from: redirect,
-                    to: configYaml['301'][redirect],
-                    permanent: true
-                });
-            }
-        } else {
-            redirects = fs.readFileSync(path.join(config.getContentPath('data'), 'redirects.json'), 'utf-8');
-            redirects = JSON.parse(redirects);
-        }
+        const filePath = getRedirectsFilePath();
+        const content = fs.readFileSync(filePath);
+        const redirects = redirectsService.settings.parseRedirectsFile(content, path.extname(filePath));
```

Comment:
Maybe instead of giving the `parseRedirectsFile` two parameters we could give it just the `filepath` and let the logic detecting the extension live inside of the `parseRedirectsFile` method?

Reply:
I've created a `loadRedirectsFile` method and removed `filePath` and `content`.